### PR TITLE
Feat/master/script to update docker images

### PR DESCRIPTION
**Pull request by anestos**

### CODE REVIEW

File: build-images.sh [hunk 1]
```diff
@@ -1,16 +1,17 @@
-#/bin/bash
+#!/bin/bash
 
-mkdir -p ${VERSION}/ubuntu
-mkdir -p ${VERSION}/centos
-mkdir -p ${VERSION}/stretch
-mkdir -p ${VERSION}/stretch-slim
+if [ ! -d "$DIRECTORY" ]; then
```
Comment (daniellindau) [inline build-images.sh]: `$DIRECTORY` doesn't seem to be used anywhere else?

File: update-images.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  echo "Downloading version $VERSION";
+  ACCESS_TOKEN=$(curl -d "grant_type=client_credentials&client_secret=${CLIENT_SECRET}&client_id=${CLIENT_ID}&scope=${SCOPE}" ${TOKEN_ENDPOINT} | jq -r '.access_token')
+
+  if [[ "$ACCESS_TOKEN" == "null" ]]; then
+    echo "Failed to get access token"
+    exit 1
+  fi
+
+  RELEASE_FILENAME=idsvr-$VERSION-linux.tar.gz
+  curl -H "Authorization: Bearer ${ACCESS_TOKEN}" ${RELEASE_API}/"${VERSION}"/linux-release > "${RELEASE_FILENAME}"
```
Comment (daniellindau) [inline update-images.sh]: Theres no need to quote the variables. `$RELEASE_API/$VERSION/linux-release` or atleast make it consistent. :)

File: build-images.sh (hunk 1 above)
Comment (anestos) [reply build-images.sh]: true, fixed

File: update-images.sh (hunk 2 above)
Comment (anestos) [reply update-images.sh]: Intellij complained for that. i ll quote both

File: build-images.sh [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
+  mkdir -p "${VERSION}/stretch-slim"
 
-cat Dockerfile-ubuntu.template | sed -e "s/{{VERSION}}/$VERSION/g" > ${VERSION}/ubuntu/Dockerfile
-cat Dockerfile-centos.template | sed -e "s/{{VERSION}}/$VERSION/g" > ${VERSION}/centos/Dockerfile
-cat Dockerfile-stretch.template | sed -e "s/{{VERSION}}/$VERSION/g" > ${VERSION}/stretch/Dockerfile
-cat Dockerfile-stretch-slim.template | sed -e "s/{{VERSION}}/$VERSION/g" > ${VERSION}/stretch-slim/Dockerfile
+  sed -e "s/{{VERSION}}/${VERSION}/g" < Dockerfile-ubuntu.template > "${VERSION}/ubuntu/Dockerfile"
+  sed -e "s/{{VERSION}}/${VERSION}/g" < Dockerfile-centos.template > "${VERSION}/centos/Dockerfile"
+  sed -e "s/{{VERSION}}/${VERSION}/g" < Dockerfile-stretch.template > "${VERSION}/stretch/Dockerfile"
+  sed -e "s/{{VERSION}}/${VERSION}/g" < Dockerfile-stretch-slim.template > "${VERSION}/stretch-slim/Dockerfile"
```
Comment (travisspencer) [inline build-images.sh]: `<` isn't needed either. sed's man page says:

> sed [-Ealn] [-e command] [-f command_file] [-i extension] [file ...]

File: update-images.sh [hunk 4]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+set -e
+
+[[ -z "${CLIENT_ID}" ]] && echo "CLIENT_ID not set" &&  exit 1;
```
Comment (travisspencer) [inline update-images.sh]: echo this to standard out? 

`echo "CLIENT_ID not set" >&2 && exit 1`

File: update-images.sh [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+TOKEN_ENDPOINT=https://login.curity.io/oauth/v2/token
+RELEASE_API=https://releaseapi.curity.io/releases
+SCOPE="release_download release_read"
+
+LATEST_RELEASE=$(find -- * -maxdepth 0 -type d | sort -rh | head -n 1)
+
+while IFS= read -r VERSION
+do
+  echo "Downloading version ${VERSION}";
+  ACCESS_TOKEN=$(curl -d "grant_type=client_credentials&client_secret=${CLIENT_SECRET}&client_id=${CLIENT_ID}&scope=${SCOPE}" "${TOKEN_ENDPOINT}" | jq -r '.access_token')
```
Comment (travisspencer) [inline update-images.sh]: probably want to pass `-s -S` to curl since you're piping it. Otherwise, the progress bar will be output.

Comment (travisspencer) [reply update-images.sh]: also, you can do `-u ${CLIENT_ID}:${CLIENT_SECRET} -d grant_type=client_credentials -d scope=$SCOPE` if you want

File: update-images.sh [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+  if [[ "${ACCESS_TOKEN}" == "null" ]]; then
+    echo "Failed to get access token"
```
Comment (travisspencer) [inline update-images.sh]: pine to standard error

**Pull request merged**